xbmc/peripherals/bus/PeripheralBus.h

```diff
@@ -54,7 +54,7 @@ namespace PERIPHERALS
     /*!
      * @return True if this bus needs to be polled for changes, false if this bus performs updates via callbacks
      */
-    bool NeedsPolling(void) const { return m_bNeedsPolling; }
+    bool NeedsPolling(void) const { CSingleLock lock(m_critSection); return m_bNeedsPolling; }
```

Comment:
It's the same basic complexity under the hood, a lock will likely be implemented using cas, switching to an atomic makes the visible parts simpler as the code would be `return m_bNeedsPolling` with no visible synchronization logic cluttering up the code